### ISSUE TITLE
Cannot run CNN with Batch Normalization from Tutorial 3 on a CPU

### ISSUE THREAD

**Issue by ScottSquash** [Microsoft/CNTK#297]
When 03_ConvBatchNorm.cntk contains:

#deviceId = 0
#imageLayout = "cudnn"
# override the above as follows when running on CPU:
deviceId = -1
imageLayout = "legacy"

I get this message:

EXCEPTION occurred: Batch normalization currently supports only cuDNN (CHW) data layout. Please specify imageLayout="cudnn" in BatchNormalization node in your NDL/BrainScript and make sure your input data layout is CHW

When 03_ConvBatchNorm.cntk contains:

deviceId = 0
imageLayout = "cudnn"
# override the above as follows when running on CPU:
#deviceId = -1
#imageLayout = "legacy"

I get this message:

EXCEPTION occurred: Default convolution engine currently supports only HWC/legacy layout.
